src/common/types/chains.ts

```diff
@@ -0,0 +1,126 @@
+import { TokenInfo } from './tokens'
+
+/**
+ * Supported chains and their `chainId` for the SDK.
+ * @enum
+ */
+export enum SupportedChainId {
+  MAINNET = 1,
+  GNOSIS_CHAIN = 100,
+  ARBITRUM_ONE = 42161,
+  BASE = 8453,
+  SEPOLIA = 11155111,
+}
+
+/**
+ * The chain id of the chain.
+ *
+ * TODO: Should we generalize it even more to allow non-EVM chains? We should probably revisit also the chain interface, and some other types.
+ */
+export type ChainId = number
+
+export interface WebUrl {
+  url: string
+  name: string
+}
+
+export type ChainRpcUrls = {
+  http: readonly string[]
+  webSocket?: readonly string[]
+}
+
+export type ChainContract = {
+  address: string
+  blockCreated?: number | undefined
+}
+
+export type ChainContracts = {
+  multicall3?: ChainContract
+  ensRegistry?: ChainContract
+  ensUniversalResolver?: ChainContract
+  universalSignatureVerifier?: ChainContract
+}
+
+/**
+ * A chain on the network.
+ *
+ * Probably we could use the viem chain definition, I think multicall3, and ensRegistry and the types defined there can be handy. But for now we are using a simplified version.
+ *
+ * For a list of chains, see: https://github.com/wevm/viem/tree/main/src/chains/definitions
+ */
+export interface ChainInfo {
+  /**
+   * The chain id.
+   */
+  readonly id: ChainId
+
+  /**
+   * Name of the chain as defined in
+   */
+  readonly name: string
+
+  /**
+   * Label of the chain. Field used for display purposes.
+   */
+  readonly label: string
```

Comment:
Do we really need both name and label? I think just label is enough

Reply:
I got it from CoW Swap. I think I can delete `name` as its used only in 2 places, doesn't look to important https://github.com/cowprotocol/cowswap/blob/main/apps/cowswap-frontend/src/common/hooks/useLegacySetChainIdToUrl.ts#L31